src/sql/workbench/contrib/query/browser/flavorStatus.ts

```diff
@@ -184,26 +193,16 @@ export class ChangeFlavorAction extends Action {
 			return this._showMessage(Severity.Info, nls.localize('alreadyConnected',
 				"A connection using engine {0} exists. To change please disconnect or change connection", currentProvider));
 		}
-		const editorWidget = getCodeEditor(activeEditor);
-		if (!editorWidget) {
-			return this._showMessage(Severity.Info, nls.localize('noEditor', "No text editor active at this time"));
-		}
 
 		// TODO #1334 use connectionManagementService.GetProviderNames here. The challenge is that the credentials provider is returned
 		// so we need a way to filter this using a capabilities check, with isn't yet implemented
-		const ProviderOptions: ISqlProviderEntry[] = [
-			new SqlProviderEntry(mssqlProviderName)
-		];
+		let ProviderOptions = this._connectionManagementService.getProviderNames().map(p => new SqlProviderEntry(p));
 
-		// TODO: select the current language flavor
 		return this._quickInputService.pick(ProviderOptions, { placeHolder: nls.localize('pickSqlProvider', "Select SQL Language Provider") }).then(provider => {
 			if (provider) {
-				activeEditor = this._editorService.activeControl;
-				const editorWidget = getCodeEditor(activeEditor);
-				if (editorWidget) {
```

Comment:
why is the `editorWidget` is not undefined check being removed?

Reply:
I removed it by mistake, I was testing something. But just using active control does not work, I am not sure why though as this is always null.
But I was looking at the code base and found that we only do getCodeEditor either on this: let activeEditor = this._editorService.activeControl.getControl(); or on this.editorService.activeTextEditorWidget. I think we should be using  let activeEditor = this._editorService.activeControl.getControl();.